Composer/packages/adaptive-form/src/components/fields/ObjectField.tsx

```diff
@@ -31,18 +33,37 @@ const ObjectField: React.FC<FieldProps<object>> = function ObjectField(props) {
 
   return (
     <React.Fragment>
-      {orderedProperties.map((row) => (
-        <FormRow
-          key={`${props.id}.${typeof row === 'string' ? row : row.join('_')}`}
-          {...rest}
-          depth={newDepth}
-          row={row}
-          schema={schema}
-          uiOptions={uiOptions}
-          value={value}
-          onChange={handleChange}
-        />
-      ))}
+      {orderedProperties.map((row) => {
+        if (typeof row === 'string' || Array.isArray(row)) {
+          return (
+            <FormRow
+              key={`${props.id}.${typeof row === 'string' ? row : row.join('_')}`}
+              {...rest}
+              definitions={definitions}
+              depth={newDepth}
+              row={row}
+              schema={schema}
+              uiOptions={uiOptions}
+              value={value}
+              onChange={handleChange}
+            />
+          );
+        } else {
```

Comment:
What is `row` when this condition is met? I would think that an additional field gets merged into the schema and resolved like every other field.

Reply:
When `row` isn't a `string` or an `array`, it is presumably an `additionalField`. I thought about adding the additional fields to the schema, and I do like how clean it would make things; however, the `onChange` handler for each individual property does not meet the requirements for some of the additional fields we currently have in the ui-plugins. For example, the `SelectSkillEndpoint` additional field needs to set multiple properties - `skillEndpoint` and `skillAppId`. Thoughts?